### ISSUE TITLE
Documentation of advanced SSL scenarios

### ISSUE THREAD

**Issue by anuraaga** [line/armeria#615]
While trying to figure out only supports ECDSA ciphers, I found it uses default netty initialization logic, which then calls native functions in tcnative where I stopped digging for now.

It would be helpful to have documentation on configuring ciphers for various scenarios, something like

1) Browser clients (large cipher suite)
2) External native clients (can be small cipher suite but should be secure)
3) Internal clients (can be just faster ciphers)

By the way, the only two ciphers my server with boringssl on linux supported are these - still haven't figured out how to support RSA ciphers like I wanted to.
```
  Supported Server Cipher(s):
Preferred TLSv1.2  256 bits  ECDHE-ECDSA-AES256-GCM-SHA384 Curve P-256 DHE 256
Accepted  TLSv1.2  128 bits  ECDHE-ECDSA-AES128-GCM-SHA256 Curve P-256 DHE 256
```

**Comment by trustin** [comment]
IIRC, there's no knobs exposed by Armeria for configuring the set of preferred ciphers. It simply uses the default cipher suite which is defined in Netty: https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2SecurityUtil.java I agree this should be fully configurable although a sensible default is good.

**Comment by anuraaga** [comment]
Yeah I noticed Http2SecurityUtil has a long list of ciphers, but when verifying with `sslscan`, I only had the two ECDSA ones, which doesn't seem like a good default. Not sure why that happens, I think it's related to this code but digging further leads a a dead link and some native function calls, quite hairy :)

https://github.com/netty/netty/blob/4.1/handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslContext.java#L287
